server/parser.go

```diff
@@ -917,15 +919,17 @@ func protoSnippet(start int, buf []byte) string {
 
 // clonePubArg is used when the split buffer scenario has the pubArg in the existing read buffer, but
 // we need to hold onto it into the next read.
-func (c *client) clonePubArg() {
+func (c *client) clonePubArg() error {
 	// Just copy and re-process original arg buffer.
 	c.argBuf = c.scratch[:0]
 	c.argBuf = append(c.argBuf, c.pa.arg...)
 
-	// This is a routed msg
-	if c.pa.account != nil {
-		c.processRoutedMsgArgs(false, c.argBuf)
-	} else {
-		c.processPub(false, c.argBuf)
+	switch c.kind {
+	case ROUTER, GATEWAY:
+		return c.processRoutedMsgArgs(false, c.argBuf)
```

Comment:
Route and Gateway have same processing. Adding LEAF but then default to processPub(). If we end-up with a client kind that does not have same msg args format, processPub will return error that will now cause a parser error.

Reply:
Yep.